sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -388,10 +388,12 @@ final class Decimal extends Ordered[Decimal] with Serializable {
       Decimal(toBigDecimal - that.toBigDecimal)
     }
   }
+  def sub(that: Decimal): Decimal = this - that
 
   // TypeCoercion will take care of the precision, scale of result
   def * (that: Decimal): Decimal =
     Decimal(toJavaBigDecimal.multiply(that.toJavaBigDecimal, MATH_CONTEXT))
+  def mul(that: Decimal): Decimal = this * that
```

Comment:
BTW I think you can call the existing operator with `.$times()` in Java code

Reply:
@srowen Thank you for this. I replaced the methods by `$minus` and `$times` as you suggested.